nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/test/java/org/apache/nifi/processors/script/TestInvokeGroovy.java

```diff
@@ -0,0 +1,280 @@
+     *
+     * @throws Exception Any error encountered while testing
+     */
+    @Test
+    public void testScriptDefinedAttribute() throws Exception {
+        InvokeScriptProcessor processor = new InvokeScriptProcessor();
+        MockProcessContext context = new MockProcessContext(processor);
+        MockProcessorInitializationContext initContext = new MockProcessorInitializationContext(processor, context);
+
+        processor.initialize(initContext);
+
+        context.setProperty(InvokeScriptProcessor.SCRIPT_ENGINE, "Groovy");
+        context.setProperty(InvokeScriptProcessor.SCRIPT_FILE, "target/test/resources/groovy/test_reader.groovy");
+        context.setProperty(InvokeScriptProcessor.MODULES, "target/test/resources/groovy");
+
+        processor.setup(context);
+
+        List<PropertyDescriptor> descriptors = processor.getSupportedPropertyDescriptors();
+        assertNotNull(descriptors);
+        assertTrue(descriptors.size() > 0);
+        boolean found = false;
+        for (PropertyDescriptor descriptor : descriptors) {
+            if (descriptor.getName().equals("test-attribute")) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue(found);
+    }
+
+    /**
+     * Tests a script that has a Groovy Processor that that reads the first line of text from the flowfiles content and
+     * stores the value in an attribute of the outgoing flowfile.
+     *
+     * @throws Exception Any error encountered while testing
+     */
+    @Test
+    public void testScriptDefinedRelationship() throws Exception {
+        InvokeScriptProcessor processor = new InvokeScriptProcessor();
+        MockProcessContext context = new MockProcessContext(processor);
+        MockProcessorInitializationContext initContext = new MockProcessorInitializationContext(processor, context);
+
+        processor.initialize(initContext);
+
+        context.setProperty(InvokeScriptProcessor.SCRIPT_ENGINE, "Groovy");
+        context.setProperty(InvokeScriptProcessor.SCRIPT_FILE, "target/test/resources/groovy/test_reader.groovy");
+
+        processor.setup(context);
+
+        Set<Relationship> relationships = processor.getRelationships();
+        assertNotNull(relationships);
+        assertTrue(relationships.size() > 0);
+        boolean found = false;
+        for (Relationship relationship : relationships) {
+            if (relationship.getName().equals("test")) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue(found);
+    }
+
+    /**
+     * Tests a script that throws a ProcessException within. The expected result is that the exception will be
+     * propagated
+     *
+     * @throws Exception Any error encountered while testing
+     */
+    @Test(expected = AssertionError.class)
+    public void testScriptException() throws Exception {
+        String scriptBody = "class GroovyProcessor implements Processor {\n" +
```

Comment:
I think we should probably move all of this into a file under src/test/resources and then read it in, in the unit test. I think it keeps the code a lot cleaner this way.